bottlecap/src/proc/mod.rs

```diff
@@ -180,6 +211,135 @@ fn get_uptime_from_path(path: &str) -> Result<f64, io::Error> {
     ))
 }
 
+#[must_use]
+pub fn get_fd_max_data(pids: &[i32]) -> f64 {
+    get_fd_max_data_from_path(PROC_PATH, pids)
+}
+
+fn get_fd_max_data_from_path(path: &str, pids: &[i32]) -> f64 {
+    let mut fd_max: f64 = 1024.0; // Default to hard limit set by AWS Lambda
+
+    for &pid in pids {
+        let limits_path = format!("{path}/{pid}/limits");
+        let Ok(file) = File::open(&limits_path) else {
+            continue;
+        };
+
+        let reader = io::BufReader::new(file);
+        for line in reader.lines().map_while(Result::ok) {
+            if line.starts_with("Max open files") {
+                let values: Vec<&str> = line.split_whitespace().collect();
+
+                // Line should have 6 items: Max open files [soft limit] [hard limit] [units]
+                if values.len() < 6 {
+                    debug!("File descriptor max data not found in file {}", limits_path);
+                    break;
+                }
+
+                // Skip past the first three items ("Max open files") to get the soft limit
+                let fd_max_pid_str = values[3];
+                let Ok(fd_max_pid) = fd_max_pid_str.parse() else {
+                    debug!("File descriptor max data not found in file {}", limits_path);
+                    break;
+                };
+                fd_max = fd_max.min(fd_max_pid);
+                break;
+            }
+        }
+    }
+
+    fd_max
+}
+
+pub fn get_fd_use_data(pids: &[i32]) -> Result<f64, io::Error> {
+    get_fd_use_data_from_path(PROC_PATH, pids)
+}
+
+fn get_fd_use_data_from_path(path: &str, pids: &[i32]) -> Result<f64, io::Error> {
+    let mut fd_use = 0;
+
+    for pid in pids {
```

Comment:
why in some loops we have the reference and in others not?
It is better to have consistency